services/token_update_nfts.proto

```diff
@@ -34,19 +34,34 @@ import "google/protobuf/wrappers.proto";
 /**
  * At consensus, updates an already created Non Fungible Token to the given values.
  * 
- * If no value is given for a field, that field is left unchanged. For an immutable tokens (that is,
- * a token without an admin key), only the expiry may be updated. Setting any other field in that
- * case will cause the transaction status to resolve to TOKEN_IS_IMMUTABLE.
+ * If no value is given for a field, that field is left unchanged.
+ * Only certain fields such as metadata can be updated. For an immutable token (that is, a token
+ * without an admin key), only the expiry may be updated. Setting any other field in that case will
+ * cause the transaction status to resolve to TOKEN_IS_IMMUTABLE.
  *
+ * Updating the metadata of an NFT does not affect its ownership or transferability.
+ * This operation is intended for updating attributes of individual NFTs in a collection.
+ *
+ * --- Signing Requirements ---
+ * 1. Whether or not a token has an admin key, its expiry can be extended with only the transaction
+ *    payer's signature.
+ * 2. To update an NFT, the account that owns the NFT must sign the transaction.
+ * 3. The metadata key is not required to be included in the TokenUpdateNftsTransactionBody as it is
+ *      not required to set the metadata. The metadata key is required to sign the metadata update.
  */
-message TokenUpdateNftTransactionBody {
+message TokenUpdateNftsTransactionBody {
+    /**
+     * The token for which to update NFTs.
+     */
+    TokenID token = 1;
+
     /**
-     * The NftID of the NFT
+     * The list of serial numbers to be updated.
      */
-    NftID nft_id = 1;
+    repeated int64 serialNumbers = 2;
```

Comment:
```suggestion
    repeated int64 serial_numbers = 2;
```

Reply:
Please do not do this. TokenService uses camel case almost everywhere. We discussed this and decided that consistency is more important.

Reply:
Disagree. We have agreed as a company to use snake_case everywhere going forward as recommended by the [Protobuf style guide](https://protobuf.dev/programming-guides/style/#message-field-names). We have not done a mass conversion everywhere but that does not mean we should introduce new areas using camel case.

Reply:
Once modularization is complete and this pbj branch is merged we should look into that mass conversion so we don't have to keep having these discussions in every PR. 😄

Reply:
When was this decided? I recall a quite recent discussion where we decided that consistency is more important, even if the outcome is sub-optimal.

Reply:
@netopyr I think initially we were using `camelCase`. But there was a decision at some point to use `snake_case`. 
You can see the latest protobufs eg : `node_stake_update.proto` all use snake_case.
Since we did not change all old ones to snake_case after that decision was made it is confusing.

Reply:
Believe it was in 2021 after a SDK team member pushed for some standardization changes via one or more [PRs](https://github.com/hashgraph/hedera-protobufs/pull/13). The decision was made on a call with leadership, though we should've documented it somewhere in a repo developer guide to make it official.